src/client/api/v1/get_balance.c

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <string.h>
+
+#include "client/api/json_utils.h"
+#include "client/api/v1/get_balance.h"
+#include "client/api/v1/response_error.h"
+#include "client/network/http.h"
+#include "core/utils/iota_str.h"
+
+res_balance_t *res_balance_new() {
+  res_balance_t *res = malloc(sizeof(res_balance_t));
+  res->is_error = false;
+  return res;
+}
+
+void res_balance_free(res_balance_t *res) {
+  if (res) {
+    if (res->is_error) {
+      res_err_free(res->u.error);
+    } else {
+      free(res->u.output_balance);
+    }
+    free(res);
+  }
+}
+
+int deser_balance_info(char const *const j_str, res_balance_t *res) {
+  char const *const key_addr = "address";
+  char const *const key_maxResults = "maxResults";
+  char const *const key_count = "count";
+  char const *const key_balance = "balance";
+  char const *const key_code = "code";
+  char const *const key_message = "message";
+  int ret = 0;
+
+  cJSON *json_obj = cJSON_Parse(j_str);
+  if (json_obj == NULL) {
+    return -1;
+  }
+
+  res_err_t *res_err = deser_error(json_obj);
+  if (res_err) {
+    // got an error response
+    res->is_error = true;
+    res->u.error = res_err;
+    ret = -1;
```

Comment:
the `ret` denote failures of this function, receiving an error response is not an error, the response status has been set in `res->is_error`, we should return 0.